webapp/plugins/jira/components/settings.jsx

```diff
@@ -0,0 +1,251 @@
+            normalizedPretext,
+            (data) => {
+                if (this.shouldCancelDispatch(normalizedPretext)) {
+                    return;
+                }
+
+                const users = Object.assign([], data.users);
+
+                AppDispatcher.handleServerAction({
+                    type: ActionTypes.SUGGESTION_RECEIVED_SUGGESTIONS,
+                    id: suggestionId,
+                    matchedPretext: normalizedPretext,
+                    terms: users.map((user) => user.username),
+                    items: users,
+                    component: UserSuggestion
+                });
+            }
+        );
+
+        return true;
+    }
+}
+
+export default class JIRASettings extends AdminSettings {
+    constructor(props) {
+        super(props);
+
+        this.getConfigFromState = this.getConfigFromState.bind(this);
+        this.renderSettings = this.renderSettings.bind(this);
+        this.handleSecretChange = this.handleSecretChange.bind(this);
+        this.handleEnabledChange = this.handleEnabledChange.bind(this);
+        this.handleUserSelected = this.handleUserSelected.bind(this);
+
+        this.userSuggestionProviders = [new UserProvider()];
+    }
+
+    getConfigFromState(config) {
+        config.PluginSettings.Plugins = {
+            jira: {
+                Enabled: this.state.enabled,
+                Secret: this.state.secret,
+                UserName: this.state.userName
+            }
+        };
+
+        return config;
+    }
+
+    getStateFromConfig(config) {
+        const settings = config.PluginSettings;
+
+        const ret = {
+            enabled: false,
+            secret: '',
+            userName: '',
+            siteURL: config.ServiceSettings.SiteURL
+        };
+
+        if (typeof settings.Plugins !== 'undefined' && typeof settings.Plugins.jira !== 'undefined') {
+            ret.enabled = settings.Plugins.jira.Enabled || settings.Plugins.jira.enabled || false;
+            ret.secret = settings.Plugins.jira.Secret || settings.Plugins.jira.secret || '';
+            ret.userName = settings.Plugins.jira.UserName || settings.Plugins.jira.username || '';
+        }
+
+        return ret;
+    }
+
+    handleSecretChange(id, secret) {
+        this.handleChange(id, secret.replace('+', '-').replace('/', '_'));
+    }
+
+    handleEnabledChange(enabled) {
+        if (enabled && this.state.secret === '') {
+            this.handleSecretChange('secret', crypto.randomBytes(256).toString('base64').substring(0, 32));
+        }
+        this.handleChange('enabled', enabled);
+    }
+
+    handleUserSelected(user) {
+        this.handleChange('userName', user.username);
+    }
+
+    renderTitle() {
+        return Utils.localizeMessage('admin.plugins.jira', 'JIRA (Beta)');
+    }
+
+    renderSettings() {
+        var webhookDocsLink = (
+            <a
+                href='https://developer.atlassian.com/jiradev/jira-apis/webhooks#Webhooks-configure'
```

Comment:
Propose we instead link to https://about.mattermost.com/default-jira-plugin

We can redirect the page either to our own docs or to https://developer.atlassian.com/jiradev/jira-apis/webhooks#Webhooks-configure.

That way if the doc page we redirect to breaks (e.g. the page was moved), all we have to fix is the redirect without changes to the code.